ext/openssl/openssl.c

```diff
@@ -3215,18 +3384,26 @@ PHP_FUNCTION(openssl_csr_new)
 						RETVAL_TRUE;
 
 						if (X509_REQ_sign(csr, req.priv_key, req.digest)) {
-							ZVAL_RES(return_value, zend_register_resource(csr, le_csr));
+							object_init_ex(return_value, x509_request_ce);
+							x509_request_obj = Z_X509_REQUEST_P(return_value);
+							x509_request_obj->csr = X509_REQ_dup(csr);
 							csr = NULL;
 						} else {
 							php_openssl_store_errors();
 							php_error_docref(NULL, E_WARNING, "Error signing request");
 						}
 
 						if (we_made_the_key) {
-							/* and a resource for the private key */
-							ZEND_TRY_ASSIGN_REF_RES(out_pkey, zend_register_resource(req.priv_key, le_key));
+							/* and an object for the private key */
+							zval *zkey_object = NULL;
+							openssl_key_object *key_object;
+							object_init_ex(zkey_object, openssl_key_ce);
+							key_object = Z_OPENSSL_KEY_P(zkey_object);
+							key_object->pkey = req.priv_key;
+
+							ZEND_TRY_ASSIGN_REF_TMP(out_pkey, zkey_object);
```

Comment:
I couldn't find out what I should do instead of `ZEND_TRY_ASSIGN_REF_RES()`. I think the current code causes a segfault.

Reply:
Have you tried using another similar macro like: ``ZEND_TRY_ASSIGN_REF_VALUE`` ?

Reply:
Yeah, I did try them :D As far as I remember it turned out that the segfault happens a few lines before, but I'm not exactly sure yet what's wrong

Reply:
Yes, AFAIK the issue is with the `key_object` being `null`, when `object_init_ex()` is invoked on it. As far as I saw it in other code, they do something like
```
zval zkey_object;
object_init_ex(&zkey_object, openssl_key_ce);
```

Although it doesn't work yet in the PR about ext/socket migration, where I'm having the same issue.

Reply:
It did work here :) There's only 1 failing test remaining. :)